services/MerchStoreService.ts

```diff
@@ -346,7 +346,9 @@ export default class MerchStoreService {
           .merchStoreCollection(txn)
           .findByUuid(updatedCollection);
         if (!collection) throw new NotFoundError('Merch collection not found');
+        updatedItem.collection = collection;
```

Comment:
If you look at line 341, there's a call to `MerchandiseItemModel.merge()`. However, line 352 (`merchItemRepository.upsertMerchItem()`) calls `MerchandiseItemModel.merge()` internally. What's the reason for this difference?

Reply:
we merge at line 341 so that we can immediately test whether the merged item is a valid item or not on line 342, although upsertMerchItem does call merge internally, it does not do the checking by itself and requires external check for validations. In addition, merge is a fast operation so we can kinda use it freely.